app/eventyay/base/payment.py

```diff
@@ -1254,7 +1254,20 @@ def checkout_prepare(self, request: HttpRequest, cart: Dict[str, Any]) -> Union[
 
         cs = cart_session(request)
         try:
-            gc = self.event.organizer.accepted_gift_cards.get(secret=request.POST.get('giftcard'))
+            gc = self.event.organizer.accepted_gift_cards.filter(secret__iexact=request.POST.get('giftcard')).first()
```

Comment:
The GiftCard code is a secret code, it is case-sensitive, we should not looking up with `iexact`.